app/src/modules/content/components/live-preview.vue

```diff
@@ -0,0 +1,247 @@
+			</span>
+			<div class="spacer" />
+			<div class="dimensions" :class="{ disabled: fullscreen }">
+				<input
+					:value="displayWidth"
+					class="width"
+					:disabled="fullscreen"
+					@input="width = Number(($event as any).target.value)"
+				/>
+				<v-icon x-small name="close" />
+				<input
+					:value="displayHeight"
+					class="height"
+					:disabled="fullscreen"
+					@input="height = Number(($event as any).target.value)"
+				/>
+				<v-select
+					v-model="zoom"
+					inline
+					:items="[
+						{ text: '25%', value: 0.25 },
+						{ text: '50%', value: 0.5 },
+						{ text: '75%', value: 0.75 },
+						{ text: '100%', value: 1 },
+						{ text: '150%', value: 1.5 },
+						{ text: '200%', value: 2 },
+					]"
+					:disabled="fullscreen"
+				/>
+			</div>
+			<v-button
+				v-tooltip.bottom.start="t('live_preview.change_size')"
+				x-small
+				icon
+				rounded
+				:secondary="fullscreen"
+				@click="toggleFullscreen"
+			>
+				<v-icon small name="devices" />
+			</v-button>
+		</div>
+		<div class="iframe-view">
+			<div
+				ref="resizeHandle"
+				class="resize-handle"
+				:style="{
+					width: width ? `${width}px` : '100%',
+					height: height ? `${height}px` : '100%',
+					resize: fullscreen ? 'none' : 'both',
+					transform: `scale(${zoom}) ${!fullscreen ? `rotateX(${rotateX}deg) rotateY(${rotateY}deg)` : ''}`,
+					'transform-origin': zoom >= 1 ? 'top left' : 'center center',
+				}"
+			>
+				<iframe
+					id="frame"
+					ref="frameEl"
+					width="100%"
+					height="100%"
+					:src="url"
+					frameborder="0"
+					@load="onIframeLoad"
+				></iframe>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+interface Props {
+	url: string;
+	inPopup?: boolean;
+}
+
+defineProps<Props>();
+
+const emit = defineEmits(['new-window']);
+
+const { t } = useI18n();
+
+const rotateX = ref<number>(0);
+const rotateY = ref<number>(0);
+const width = ref<number>();
+const height = ref<number>();
+const zoom = ref<number>(1);
+const displayWidth = ref<number>();
+const displayHeight = ref<number>();
+const isRefreshing = ref<boolean>(false);
+
+const resizeHandle = ref<HTMLDivElement>();
+
+const fullscreen = computed(() => {
+	return width.value === undefined && height.value === undefined;
+});
+
+function toggleFullscreen() {
+	if (fullscreen.value) {
+		width.value = 400;
+		height.value = 600;
+	} else {
+		width.value = undefined;
+		height.value = undefined;
+		zoom.value = 1;
+	}
+}
+
+const frameEl = ref<HTMLIFrameElement>();
+
+function refresh() {
+	if (!frameEl.value) return;
+
+	// this is technically a self-assignment, but it works to refresh the iframe
+	const newSrc = frameEl.value.src;
+	frameEl.value.src = newSrc;
+	isRefreshing.value = true;
+}
+
+function onIframeLoad() {
+	isRefreshing.value = false;
+}
+
+(window as any).refreshLivePreview = refresh;
```

Comment:
Instead of typecasting custom properties can be added to the window like so in TS:
```suggestion
declare global {
	interface Window {
		refreshLivePreview: () => void;
	}
}

window.refreshLivePreview = refresh;
```

Reply:
Good catch! Seems like there's actually the same approach for another custom property in:

https://github.com/directus/directus/blob/4f768ca346fd4ce12c08884bf90de09b22cb4bdb/packages/composables/src/use-element-size.ts#L5-L9

implemented the suggested change in https://github.com/directus/directus/pull/17794/commits/4f768ca346fd4ce12c08884bf90de09b22cb4bdb.

Reply:
Awesome don't forget to clean up the remaining `(window as any)` references in the source 🚀 😄